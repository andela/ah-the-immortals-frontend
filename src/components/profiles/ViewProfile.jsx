import React from 'react';
import PropTypes from 'prop-types';
import { Container } from 'react-bootstrap';
import profileImage from '../../services/images/profile.png';
import EditProfile from '../../views/profiles/EditProfile';
import Footer from '../home/HomeFooter';
import isLoggedIn from '../../services/checkAuthentication';
import createBrowserHistory from '../../services/history';

const history = createBrowserHistory; 
export const ProfileView = (props) => {
  const { username, firstName, lastName, bio, image } = props;
  const loggedIn= isLoggedIn();
  if(!loggedIn){
    history.push('/');
  }
  return (
    <div>
      <Container className="profile">
        <div className="container-fluid">
          <div className="container">
            <div className="card profile-card">
              <div className="centered">
                <div className="float-left profile-avatar">
                  <img src={image || profileImage} alt="Avatar" className="avatar" />
                </div>
              </div>
              <h4>
                @
                {
                  `${username || 'username'} [${firstName} ${lastName}]`
                }
              </h4>
              <p>{bio || 'This is my biography'}</p>
            </div>
          </div>
          <ul className="nav nav-tabs">
            <li className="nav-item">
              <a className="nav-link active" data-toggle="tab" href="#home">
                <b> 3 </b> 
              Articles
              </a>
            </li>
            <li className="nav-item">
              <a className="nav-link" data-toggle="tab" href="#profile">Edit Profile</a>
            </li>
            <li className="nav-item">
              <a className="nav-link" data-toggle="tab" href="#followers">
                <b> 122 </b>
              Followers
              </a>
            </li>
            <li className="nav-item">
              <a className="nav-link" data-toggle="tab" href="#following">
                <b> 200 </b> 
              Following
              </a>
            </li>
          </ul>
          <div id="myTabContent" className="tab-content profile-content-articles">
            <div className="tab-pane fade active show" id="home">
              <div className="row">
                <div className="col">
                  <div className="card">
                    <img className="card-img-top" src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg" alt="article" />
                    <div className="card-body">
                      <h4 className="card-title"><a href="#card">Article title</a></h4>
                      <p className="card-text">Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.</p>
                      <a href="#view" className="btn btn-primary">View</a>
                    </div>
                  </div>
                </div>
                <div className="col">
                  <div className="card">
                    <img className="card-img-top" src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg" alt="article" />
                    <div className="card-body">
                      <h4 className="card-title"><a href="#card">Article title</a></h4>
                      <p className="card-text">Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.</p>
                      <a href="#view" className="btn btn-primary">View</a>
                    </div>
                  </div>
                </div>
                <div className="col">
                  <div className="card">
                    <img className="card-img-top" src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg" alt="article" />
                    <div className="card-body">
                      <h4 className="card-title"><a href="#card">Article title</a></h4>
                      <p className="card-text">Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.</p>
                      <a href="#view" className="btn btn-primary">View</a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className="tab-pane fade" id="profile">
              <EditProfile />
            </div>
          </div>
        </div>
      </Container>
      <Footer />
    </div>
  );
};


ProfileView.propTypes = {
  username: PropTypes.string,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  image: PropTypes.string,
  bio: PropTypes.string,
};

ProfileView.defaultProps = {
  username: '',
  firstName: '',
  lastName: '',
  image: '',
  bio: '',
};

export default ProfileView;

