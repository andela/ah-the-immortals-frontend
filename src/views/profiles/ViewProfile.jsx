import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ProfileView from '../../components/profiles/ViewProfile';
import { fetchProfile, editProfile } from '../../redux/actions/profile.actions';
import { userFollowingAction, userFollowersAction } from '../../redux/actions/userFollow.action';

class ViewProfile extends Component {
  state = {};
  componentDidMount() {
    const { fetchProfile: Profile, userFollowingAction: userFollowing, userFollowersAction: userFollowers } = this.props;
    Profile(this.state);
    userFollowers();
    userFollowing();
  }
  render() {
    const { profile: { username, bio, img_url, first_name, last_name }, data: { userCount }, users: { count } } = this.props;
    return (
      <ProfileView
        username={username}
        image={img_url}
        bio={bio}
        firstName={first_name}
        lastName={last_name}
        following={userCount}
        followers={count}
      />
    );
  }
}

ViewProfile.propTypes = {
  fetchProfile: PropTypes.func.isRequired,
  profile: PropTypes.shape({}),
  userFollowingAction: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
  userFollowersAction: PropTypes.func.isRequired,
  users: PropTypes.object.isRequired
};

ViewProfile.defaultProps = {
  profile: {}
};

const mapStateToProps = state => ({
  profile: state.Profile.profile,
  data: state.following,
  users: state.followers
});

export default connect(mapStateToProps, { fetchProfile, editProfile, userFollowingAction, userFollowersAction })(ViewProfile);
