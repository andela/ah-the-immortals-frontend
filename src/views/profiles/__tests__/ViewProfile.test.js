import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ViewProfile from '../ViewProfile';

describe('Test view profile container', () => {
  let testStore = configureMockStore([thunk]);
  let store = testStore(
    {
      Profile: {profile: {} }, 
      following: {
        data: [
          {
            'username': 'slim',
            'first_name': '',
            'last_name': '',
            'bio': '',
            'img_url': 'https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g',
            'created_at': '2019-05-30 11:52:18',
            'updated_at': '2019-06-13 17:59:21',
            'following': true
          }
        ],
        errors: {},
        count: 0
      },
      followers: {
        users: [
          {
            'username': 'slim',
            'first_name': '',
            'last_name': '',
            'bio': '',
            'img_url': 'https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g',
            'created_at': '2019-05-30 11:52:18',
            'updated_at': '2019-06-13 17:59:21',
            'following': true
          }
        ],
        error: {},
        userCount: 0
      }
    });

  
  const wrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <ViewProfile />
      </BrowserRouter>
    </Provider>
  );

  it('test view profile container snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
