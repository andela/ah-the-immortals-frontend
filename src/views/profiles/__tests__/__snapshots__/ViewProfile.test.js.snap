// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view profile container test view profile container snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(ViewProfile)>
        <ViewProfile
          data={
            Object {
              "count": 0,
              "data": Array [
                Object {
                  "bio": "",
                  "created_at": "2019-05-30 11:52:18",
                  "first_name": "",
                  "following": true,
                  "img_url": "https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g",
                  "last_name": "",
                  "updated_at": "2019-06-13 17:59:21",
                  "username": "slim",
                },
              ],
              "errors": Object {},
            }
          }
          editProfile={[Function]}
          fetchProfile={[Function]}
          profile={Object {}}
          userFollowersAction={[Function]}
          userFollowingAction={[Function]}
          users={
            Object {
              "error": Object {},
              "userCount": 0,
              "users": Array [
                Object {
                  "bio": "",
                  "created_at": "2019-05-30 11:52:18",
                  "first_name": "",
                  "following": true,
                  "img_url": "https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g",
                  "last_name": "",
                  "updated_at": "2019-06-13 17:59:21",
                  "username": "slim",
                },
              ],
            }
          }
        >
          <ProfileView
            bio=""
            firstName=""
            followers={0}
            following={0}
            image=""
            lastName=""
            username=""
          >
            <div>
              <Container
                className="profile"
                fluid={false}
              >
                <div
                  className="profile container"
                >
                  <div
                    className="container-fluid"
                  >
                    <div
                      className="container"
                    >
                      <div
                        className="card profile-card"
                      >
                        <div
                          className="centered"
                        >
                          <div
                            className="float-left profile-avatar"
                          >
                            <img
                              alt="Avatar"
                              className="prof-avatar"
                              src="test-file-stub"
                            />
                          </div>
                        </div>
                        <h4>
                          @
                          username [ ]
                        </h4>
                        <p>
                          This is my biography
                        </p>
                      </div>
                    </div>
                    <ul
                      className="nav nav-tabs"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link active"
                          data-toggle="tab"
                          href="#home"
                        >
                          <b>
                             3 
                          </b>
                          Articles
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          data-toggle="tab"
                          href="#profile"
                        >
                          Edit Profile
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          data-toggle="tab"
                          href="#followers"
                        >
                          <b>
                            0 
                          </b>
                          Followers
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          data-toggle="tab"
                          href="#following"
                        >
                          <b>
                            0 
                          </b>
                          Following
                        </a>
                      </li>
                    </ul>
                    <div
                      className="tab-content profile-content-articles"
                      id="myTabContent"
                    >
                      <div
                        className="tab-pane fade active show"
                        id="home"
                      >
                        <div
                          className="row"
                        >
                          <div
                            className="col"
                          >
                            <div
                              className="card"
                            >
                              <img
                                alt="article"
                                className="card-img-top"
                                src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg"
                              />
                              <div
                                className="card-body"
                              >
                                <h4
                                  className="card-title"
                                >
                                  <a
                                    href="#card"
                                  >
                                    Article title
                                  </a>
                                </h4>
                                <p
                                  className="card-text"
                                >
                                  Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.
                                </p>
                                <a
                                  className="btn btn-primary"
                                  href="#view"
                                >
                                  View
                                </a>
                              </div>
                            </div>
                          </div>
                          <div
                            className="col"
                          >
                            <div
                              className="card"
                            >
                              <img
                                alt="article"
                                className="card-img-top"
                                src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg"
                              />
                              <div
                                className="card-body"
                              >
                                <h4
                                  className="card-title"
                                >
                                  <a
                                    href="#card"
                                  >
                                    Article title
                                  </a>
                                </h4>
                                <p
                                  className="card-text"
                                >
                                  Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.
                                </p>
                                <a
                                  className="btn btn-primary"
                                  href="#view"
                                >
                                  View
                                </a>
                              </div>
                            </div>
                          </div>
                          <div
                            className="col"
                          >
                            <div
                              className="card"
                            >
                              <img
                                alt="article"
                                className="card-img-top"
                                src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg"
                              />
                              <div
                                className="card-body"
                              >
                                <h4
                                  className="card-title"
                                >
                                  <a
                                    href="#card"
                                  >
                                    Article title
                                  </a>
                                </h4>
                                <p
                                  className="card-text"
                                >
                                  Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.
                                </p>
                                <a
                                  className="btn btn-primary"
                                  href="#view"
                                >
                                  View
                                </a>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className="tab-pane fade"
                        id="profile"
                      >
                        <Connect(ProfileContainer)>
                          <ProfileContainer
                            editProfile={[Function]}
                            profile={Object {}}
                          >
                            <div
                              className="container profile-form"
                            >
                              <ProfileForm
                                handleChangeProfile={[Function]}
                                handleFileUpload={[Function]}
                                handleSubmit={[Function]}
                              >
                                <div>
                                  <form
                                    onSubmit={[Function]}
                                  >
                                    <div
                                      className="form-group"
                                    >
                                      <label
                                        htmlFor="firstname"
                                      >
                                        First name
                                        <br />
                                        <input
                                          className="form-control"
                                          name="first_name"
                                          onChange={[Function]}
                                          type="text"
                                        />
                                      </label>
                                    </div>
                                    <div
                                      className="form-group"
                                    >
                                      <label
                                        htmlFor="lastname"
                                      >
                                        Last name
                                        <br />
                                        <input
                                          className="form-control"
                                          name="last_name"
                                          onChange={[Function]}
                                          type="text"
                                        />
                                      </label>
                                    </div>
                                    <div
                                      className="form-group"
                                    >
                                      <label
                                        htmlFor="bio"
                                      >
                                        Bio
                                        <br />
                                        <textarea
                                          className="form-control"
                                          name="bio"
                                          onChange={[Function]}
                                        />
                                      </label>
                                    </div>
                                    <div
                                      className="btn-group mr-2"
                                      role="group"
                                    >
                                      <button
                                        className="btn btn-primary"
                                        id="save"
                                        type="submit"
                                      >
                                        Save
                                      </button>
                                    </div>
                                    <div
                                      className="btn-group"
                                      role="group"
                                    >
                                      <button
                                        className="btn btn-primary"
                                        onClick={[Function]}
                                        type="submit"
                                      >
                                        Upload Photo
                                      </button>
                                    </div>
                                  </form>
                                </div>
                              </ProfileForm>
                            </div>
                          </ProfileContainer>
                        </Connect(ProfileContainer)>
                      </div>
                      <div
                        className="tab-pane fade"
                        id="following"
                      >
                        <Connect(FollowingContainer)>
                          <FollowingContainer
                            data={
                              Object {
                                "count": 0,
                                "data": Array [
                                  Object {
                                    "bio": "",
                                    "created_at": "2019-05-30 11:52:18",
                                    "first_name": "",
                                    "following": true,
                                    "img_url": "https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g",
                                    "last_name": "",
                                    "updated_at": "2019-06-13 17:59:21",
                                    "username": "slim",
                                  },
                                ],
                                "errors": Object {},
                              }
                            }
                            userFollowingAction={[Function]}
                          >
                            <div>
                              <Following
                                bio=""
                                created_at="2019-05-30 11:52:18"
                                first_name=""
                                following={true}
                                img_url="https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g"
                                key="slim"
                                last_name=""
                                updated_at="2019-06-13 17:59:21"
                                username="slim"
                              >
                                <div>
                                  <div
                                    className="user-card"
                                  >
                                    <div
                                      className="follow-card list-card"
                                    >
                                      <img
                                        alt="Profile pic"
                                        className="user-avatar float-left"
                                        src="test-file-stub"
                                      />
                                      <div
                                        className="user-data"
                                      >
                                        <h6>
                                          <Link
                                            to="/profile/slim"
                                          >
                                            <a
                                              href="/profile/slim"
                                              onClick={[Function]}
                                            >
                                              slim
                                            </a>
                                          </Link>
                                        </h6>
                                        <p />
                                      </div>
                                    </div>
                                    <hr />
                                  </div>
                                </div>
                              </Following>
                            </div>
                          </FollowingContainer>
                        </Connect(FollowingContainer)>
                      </div>
                      <div
                        className="tab-pane fade"
                        id="followers"
                      >
                        <Connect(FollowersContainer)>
                          <FollowersContainer
                            userFollowersAction={[Function]}
                            users={
                              Object {
                                "error": Object {},
                                "userCount": 0,
                                "users": Array [
                                  Object {
                                    "bio": "",
                                    "created_at": "2019-05-30 11:52:18",
                                    "first_name": "",
                                    "following": true,
                                    "img_url": "https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g",
                                    "last_name": "",
                                    "updated_at": "2019-06-13 17:59:21",
                                    "username": "slim",
                                  },
                                ],
                              }
                            }
                          >
                            <div>
                              <Followers
                                bio=""
                                created_at="2019-05-30 11:52:18"
                                first_name=""
                                following={true}
                                img_url="https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g"
                                key="slim"
                                last_name=""
                                updated_at="2019-06-13 17:59:21"
                                username="slim"
                              >
                                <div>
                                  <div
                                    className="user-card"
                                  >
                                    <div
                                      className="follow-card list-card"
                                    >
                                      <img
                                        alt="Profile pic"
                                        className="user-avatar float-left"
                                        src="https://res.cloudinary.com/grean/image/upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g"
                                      />
                                      <div
                                        className="user-data"
                                      >
                                        <h6>
                                          <Link
                                            to="/profile/slim"
                                          >
                                            <a
                                              href="/profile/slim"
                                              onClick={[Function]}
                                            >
                                              slim
                                            </a>
                                          </Link>
                                        </h6>
                                        <p />
                                      </div>
                                    </div>
                                    <hr />
                                  </div>
                                </div>
                              </Followers>
                            </div>
                          </FollowersContainer>
                        </Connect(FollowersContainer)>
                      </div>
                    </div>
                  </div>
                </div>
              </Container>
              <Footer>
                <footer
                  className="bg-light text-dark mt-4 fixed-bottom"
                >
                  <div
                    className="container-fluid py-2"
                  >
                    <div
                      className="row"
                    >
                      <div
                        className="col"
                      >
                        <span
                          className="small"
                        >
                          <br />
                          <Link
                            to="#null"
                          >
                            <a
                              href="/#null"
                              onClick={[Function]}
                            >
                              Terms and Conditions
                            </a>
                          </Link>
                        </span>
                      </div>
                      <div
                        className="col text-center"
                      >
                        <br />
                        Authors Haven
                         
                      </div>
                      <div
                        className="col text-right small align-self-end"
                      >
                        Copyright ©
                         
                        2019
                         
                        Immortals
                      </div>
                    </div>
                  </div>
                </footer>
              </Footer>
            </div>
          </ProfileView>
        </ViewProfile>
      </Connect(ViewProfile)>
    </Router>
  </BrowserRouter>
</Provider>
`;
